buzhash.go

```diff
@@ -0,0 +1,115 @@
+package chunk
+
+import (
+	"io"
+	"math/bits"
+
+	pool "github.com/libp2p/go-buffer-pool"
+)
+
+const (
+	buzMin  = 128 << 10
+	buzMax  = 512 << 10
```

Comment:
Can we configure these? Can we configure the expected/average chunk size?

Reply:
I will check what is the perf penalty for making these configurable.

Reply:
6%:
```
name       old time/op    new time/op    delta
Buzhash-4    14.2ms ± 7%    15.1ms ± 6%   +6.36%  (p=0.000 n=20+19)

name       old speed      new speed      delta
Buzhash-4  1.18GB/s ± 6%  1.11GB/s ±14%   -6.64%  (p=0.000 n=20+20)
```
Your call

Reply:
That's annoying. I'm fine leaving that off for now.